drivers/wifi/nrf700x/osal/fw_if/umac_if/src/tx.c

```diff
@@ -1311,41 +1253,69 @@ enum wifi_nrf_status tx_init(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx)
 		wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
 				      "%s: Unable to allocate Wakeup Client List\n",
 				      __func__);
+		goto tx_spin_lock_free;
+	}
 
-		wifi_nrf_osal_spinlock_free(fmac_dev_ctx->fpriv->opriv,
-					    fmac_dev_ctx->tx_config.tx_lock);
-
-		wifi_nrf_osal_mem_free(fmac_dev_ctx->fpriv->opriv,
-				       fmac_dev_ctx->tx_config.buf_pool_bmp_p);
-
-		for (i = 0; i < fpriv->num_tx_tokens; i++) {
-			wifi_nrf_utils_list_free(fpriv->opriv,
-						 fmac_dev_ctx->tx_config.pkt_info_p[i].pkt);
-		}
+	fmac_dev_ctx->twt_sleep_status = WIFI_NRF_FMAC_TWT_STATE_AWAKE;
 
-		wifi_nrf_osal_mem_free(fmac_dev_ctx->fpriv->opriv,
-				       fmac_dev_ctx->tx_config.pkt_info_p);
+#ifdef CONFIG_NRF700X_TX_DONE_WQ_ENABLED
+	fmac_dev_ctx->tx_done_tasklet = wifi_nrf_osal_tasklet_alloc(fpriv->opriv,
+								    WIFI_NRF_TASKLET_TYPE_TX_DONE);
+	if (!fmac_dev_ctx->tx_done_tasklet) {
+		wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: Unable to allocate tx_done_tasklet\n",
+				      __func__);
+		goto wakeup_client_q_free;
+	}
+	fmac_dev_ctx->tx_config.tx_done_tasklet_event_q = wifi_nrf_utils_q_alloc(fpriv->opriv);
+	if (!fmac_dev_ctx->tx_config.tx_done_tasklet_event_q) {
+		wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: Unable to allocate tx_done_tasklet_event_q\n",
+				      __func__);
+		goto tx_done_tasklet_free;
+	}
 
-		for (i = 0; i < WIFI_NRF_FMAC_AC_MAX; i++) {
-			for (j = 0; j < MAX_SW_PEERS; j++) {
-				q_ptr = fmac_dev_ctx->tx_config.data_pending_txq[j][i];
+	wifi_nrf_osal_tasklet_init(fmac_dev_ctx->fpriv->opriv,
+				   fmac_dev_ctx->tx_done_tasklet,
+				   tx_done_tasklet_fn,
+				   (unsigned long)fmac_dev_ctx);
+#endif /* CONFIG_NRF700X_TX_DONE_WQ_ENABLED */
+	return WIFI_NRF_STATUS_SUCCESS;
+#ifdef CONFIG_NRF700X_TX_DONE_WQ_ENABLED
+tx_done_tasklet_free:
+	wifi_nrf_osal_tasklet_free(fpriv->opriv,
+				   fmac_dev_ctx->tx_done_tasklet);
+wakeup_client_q_free:
+	wifi_nrf_utils_q_free(fpriv->opriv, fmac_dev_ctx->tx_config.wakeup_client_q);
+#endif /* CONFIG_NRF700X_TX_DONE_WQ_ENABLED */
+tx_spin_lock_free:
+	wifi_nrf_osal_spinlock_free(fmac_dev_ctx->fpriv->opriv,
+					fmac_dev_ctx->tx_config.tx_lock);
```

Comment:
Indentation

Reply:
Tried again, aligning to open brace vs. not exceeding 100 lines vs. using tabs as much as possible, its impossible to do this alignment anyother (AFAIK), so, closing this comment as checkpatch is passing.